### PR TITLE
docs: docs style guide + contribution guide reorganization

### LINKED ISSUE #3215: add style guide for documentation pages
> Please add all formatting things (heading 1's can't be used, we're going with - instead of * or +) in the guide for contributing docs.

_Originally posted by @RoyalFox2140 in https://github.com/cataclysmbnteam/Cataclysm-BN/issues/3190#issuecomment-1731969054_
            
```[tasklist]
### Tasks
- [x] use `-` for unordered lists
- [x] prefer using aside syntax
- [x] add caveats on aside syntax
- [x] autoformat markdown with deno
- [x] avoid using h1(`#`)
- [x] front matters
```

### PR DESCRIPTION
**Pull request by scarf005**
## Summary

SUMMARY: Infrastructure "Added style guide for writing new docs, reorganized contribution guide"

## Purpose of change

- fix #3215 

## Describe the solution

- organized `contributing` section into 4
- fixed `scripts/front_matter.ts` adding extra newlines
- rewrote `issues.md` to better fit BN
- reordered image files as it was hard to access images from other langs in current location

## Testing

image

### CODE REVIEW

**Review by olanti-p — CHANGES_REQUESTED** [review]
Title card for contributing to docs gives 404
image

File: .github/labeler.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ translation:
   - lang/**/*
 
 docs:
+  - doc/**/*
   - docs/**/*
```
[the remark this answers is not in the document]
Comment (scarf005) [reply .github/labeler.yml:18]: remnant of initial docs page PR when it began on `docs/`. removed.

File: doc/src/content/docs/en/contribute/explanation/docs_style.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+---
+
+## [Asides](https://starlight.astro.build/guides/authoring-content/#asides)
+
+### Newlines
+
+there should be a newline between the asides and the content, otherwise the formatter will merge
+them into one line, and the asides will not be rendered correctly.
+
+:::caution Avoid this :::
```
[the remark this answers is not in the document]
Comment (scarf005) [reply doc/src/content/docs/en/contribute/explanation/docs_style.md:158]: image

an example, but poorly demonstrated. reworded a bit.

File: doc/src/content/docs/en/contribute/explanation/pr_style.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Example: `SUMMARY: Content "Adds new mutation category 'Mouse'"`
+
+See the [Changelog Guidelines](../reference/changelog_guidelines.md) for explanations of the
+categories.
+
+### Closing issues using keywords
+
+One more thing: when marking your PR as closing, fixing, or resolving issues, please include this
+somewhere in the description:
```
[the remark this answers is not in the document]
Comment (scarf005) [reply doc/src/content/docs/en/contribute/explanation/pr_style.md:34]: image

done.

**Review by chaosvolt — COMMENTED** [review]
The only real feedback I can offer here is I still dislike how painful it is to find a given md file in the docs folder manually, which isn't relevant to this PR since I should've spotted that and remarked on it when it was first restructured.

**Comment by scarf005** [comment]
> The only real feedback I can offer here is I still dislike how painful it is to find a given md file in the docs folder manually, which isn't relevant to this PR since I should've spotted that and remarked on it when it was first restructured.

after 4 months now i get how painful the docs layout have become. Closing as I'm thinking of overhauling the site with structure which is better than `doc/src/content/docs/en/contribute/changelog_guidelines.md`.

**Pull request closed without merging**